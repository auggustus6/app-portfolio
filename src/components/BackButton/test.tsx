/* eslint-disable no-undef */

import React from 'react';
import BackButon from '.';
import { fireEvent, render } from '../../utils/tests/helpers';

const backIcon = require('../../assets/images/chevron_left_black.png');

const mockedDispatch = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      goBack: mockedDispatch,
    }),
  };
});

it('should render correctly', async () => {
  const { toJSON } = render(<BackButon backIcon={backIcon} />);
  expect(toJSON()).toMatchSnapshot();
});

it('should call the navigation function', async () => {
  const { getByRole } = render(<BackButon backIcon={backIcon} />);
  const btnElement = getByRole('button');

  fireEvent.press(btnElement);
  expect(mockedDispatch).toBeCalled();
});

it('should render some image source', async () => {
  const { getByRole } = render(<BackButon backIcon={backIcon} />);
  const imageElement = getByRole('image');

  expect(imageElement.props.source).toBe(backIcon);
});
