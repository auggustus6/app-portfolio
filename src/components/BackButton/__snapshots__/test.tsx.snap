// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "left": 20,
      "top": 100,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "left": 30,
      "opacity": 1,
      "position": "absolute",
      "top": 30,
      "zIndex": 9,
    }
  }
>
  <Image
    accessibilityRole="image"
    source={
      Object {
        "testUri": "../../../src/assets/images/chevron_left_black.png",
      }
    }
    style={
      Array [
        Object {
          "height": 98,
          "width": 98,
        },
      ]
    }
  />
</View>
`;
